sesman/session.c

```diff
@@ -173,6 +169,7 @@ x_server_running_check_ports(int display)
 
     if (!x_running)
     {
+        LOG(LOG_LEVEL_INFO, "Did not find a running X server at %s", text);
```

Comment:
This is going to potentially generate a lot of INFO messages, as `x_server_running_check_ports()` is called in a loop, and a message is logged for each reason why the X server isn't running. You're already logging the reason why the X server isn't running at the end of the function, so I think you could remove the "Did not find..." messages and just beef up the logging at the end (i.e.):-

```c
if (x_running)
{
    if (isdigit(text[0]))
    {
        LOG(LOG_LEVEL_INFO, "Found X server active on display %d (TCP port:%s)", display, text);
    }
    else
    {
        LOG(LOG_LEVEL_INFO, "Found X server active on display %d (file:%s)", display, text);
    }
}
else
{
    LOG(LOG_LEVEL_INFO, "No X server seems to be active on display %d", display);
}
```

That can be somewhat improved on (particularly using `isdigit` to detect a port), but I hope it gives you the idea.

Reply:
The logging at the end of the function only says that last port/socket that was checked for an X server. In my past debugging I have found it very useful in the past to see all of the ports/sockets where xrdp is looking for an X server and in which order the checks are performed. So I've kept all the log messages but lowered the logging level to debug so that it's less noisy under normal circumstances.